src/code-templates/libraries/logger/index.ts

```diff
@@ -2,44 +2,43 @@ import { Logger, LoggerConfiguration } from './definition';
 import PinoLogger from './pino.logger';
 
 export class LoggerWrapper implements Logger {
-  private underlyingLogger: Logger | null = null;
+  #underlyingLogger: Logger | null = null;
 
   configureLogger(
     configuration: Partial<LoggerConfiguration>,
     overrideIfExists = true
   ): void {
-    if (this.underlyingLogger === null || overrideIfExists === true) {
-      this.underlyingLogger = new PinoLogger(
+    if (this.#underlyingLogger === null || overrideIfExists === true) {
+      this.#underlyingLogger = new PinoLogger(
         configuration.level || 'info',
-        configuration.prettyPrint || false,
-        undefined
+        configuration.prettyPrint || false
       );
     }
   }
 
   resetLogger() {
-    this.underlyingLogger = null;
+    this.#underlyingLogger = null;
   }
 
-  debug(message: string, ...args: unknown[]): void {
+  debug(message: string, metadata?: Record<any, any>): void {
     this.configureLogger({}, false);
-    this.underlyingLogger?.debug(message, ...args);
+    this.#underlyingLogger!.debug(message, metadata);
```

Comment:
Minor: Shall we maybe include getUnderlyingLogger function that will throw and stdout if logger instance does not exist? Theoretically, the user might call resetLogger + info

```getUnderlyingLogger().info(message, metadata);```

Reply:
we call configure logger in line 24

Reply:
What guarantee you have the it will get called? There should be no trust between order-service and logger

Also, if you put this function, no need to to put "!" in every call to logger - this function will return a guaranteed instance of logger

Your call, don't pause because of this

Reply:
I need to put `!` anyway as the property is nullable

I'm sure the configure is getting called as I call this in each logging method (i.e inside info, warn and etc...)